File: Arrays/Lucky-Numbers-in-a-Matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {

        List<Integer> ans = new ArrayList<>();

        for(int i=0; i< matrix.length; i++){
            int k = 0;

            int small = Integer.MAX_VALUE;
            for(int j=0; j< matrix[0].length; j++){
                if(small > matrix[i][j]){
                    small = matrix[i][j];
                    k = j;
                }
            }

            int big = Integer.MIN_VALUE;

            for(int j=0; j< matrix.length; j++){
                big = Math.max(matrix[j][k], big);
            }

            if(big == small){
                ans.add(big);
            }
        }
        return ans;
    }
}

File: Arrays/Lucky-Numbers-in-a-Matrix.py
class Solution:
    def luckyNumbers(self, matrix: List[List[int]]) -> List[int]:

        ans = []

        for i in range(0, len(matrix)):
            k = 0
            small = sys.maxsize
            for j in range(0, len(matrix[0])):
                if small > matrix[i][j]:
                    small = matrix[i][j]
                    k = j

            big = -sys.maxsize

            for j in range(len(matrix)):
                big = max(matrix[j][k], big)

            if big == small:
                ans.append(big)

        return ans
